modules/programs/swaylock.nix

```diff
@@ -16,7 +16,21 @@ in {
         false otherwise
       '';
       example = true;
-      description = "Whether to enable swaylock.";
+      description = ''
+        Whether to enable swaylock.
+
+        Note that PAM must be configured to enable swaylock to perform
+        authentication. The package installed through home-manager
+        will *not* be able to unlock the session without this
+        configuration.
+
+        On NixOS, this is by default enabled with the sway module, but
+        for other compositors it can currently be enabled using:
+
+        ```
+        security.pam.services.swaylock = {};
```

Comment:
Thanks! In reply to your comment on #4411 that Home Manager cannot set this option: in Home Manager's [`nixos/default.nix`](https://github.com/nix-community/home-manager/blob/19c6a4081b14443420358262f8416149bd79561a/nixos/default.nix#L21) file, the `sharedModules` option does actually configure the system-wide NixOS options in the case that the Home Manager NixOS module is used. Do you think it would be a good idea to add `security.pam.services.swaylock = {};` there?

Reply:
No worries! home-manager + NixOS integration like this would actually be interesting when/if they are ever merged into nixpkgs. But yeah, for now I don't think it's actually possible.

Reply:
As a NixOS module, HM _could_ do it, but seeing now that it does not tweak the system configuration other than to set up a service for each user, I wonder if that's on purpose to avoid surprising the user...

Reply:
I doubt edge cases like this one were ever actually considered, but yeah, I think home-manager tries to stick to user configuration as much as possible intentionally. The NixOS module is just so you can easily match nixpkgs versions and don't need to run two commands when rebuilding your system.

If home-manager were merged into nixpkgs, it might be interesting to have tighter coupling. It could lead to worse behavior on non-NixOS systems over time though, as these kinds of things would be tested less, so maybe it wouldn't be the best idea anyway. Either way, food for a later discussion ;)